deploy-operator/Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Build the manager binary
+FROM golang:1.13 as builder
+
+WORKDIR /workspace
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
+
+# Copy the go source
+COPY main.go main.go
```

Comment:
This seems like it could lead to headaches if we add additional files in the root directory, or additional subdirectories. I'd recommend just `COPY . .`

Reply:
I'm not sure I want to do this. 
This project comes with a ton of random files here and there and it feels wrong to just bring them into the dockerfile if they're not necessary. What do you think?

I'm not 100% sure on what I'm saying either though, so feel free to override this!

Reply:
This Dockerfile uses the builder pattern, so the final image that gets shipped contains only the binary and some minimal dependencies. Copying the random files in here won't affect the final output. Given that most people's mental model of the docker build is "it builds based on the files in this directory," rather than "it builds based on the subset of files listed in Dockerfile," I think copying everything is worthwhile.

Reply:
Sounds good to me then! Thanks for that explanation :)